packages/@sanity/base/src/components/validation/validationList.tsx

```diff
@@ -0,0 +1,92 @@
+import React, {useCallback} from 'react'
+import {
+  ObjectSchemaType,
+  Path,
+  Marker,
+  isValidationErrorMarker,
+  isValidationWarningMarker,
+} from '@sanity/types'
+import styled from 'styled-components'
+import {ListItem} from './listItem'
+
+interface ValidationListProps {
+  documentType?: ObjectSchemaType
+  kind?: 'simple'
+  markers: Marker[]
+  onFocus?: (path: Path) => void
+  onClose?: () => void
+  // showLink?: boolean
+  truncate?: boolean
+}
+
+const ListContainer = styled.div`
+  max-width: 250px;
+  max-height: 60vh;
+  overflow: auto;
+`
+
+export function ValidationList(props: ValidationListProps) {
+  const {
+    documentType,
+    kind,
+    markers,
+    onFocus,
+    onClose,
+    truncate,
+    // showLink,
+  } = props
+
+  const errors = markers.filter(isValidationErrorMarker)
+  const warnings = markers.filter(isValidationWarningMarker)
+
+  const handleClick = useCallback(
+    (path: Path = []) => {
+      if (onFocus) onFocus(path)
+      if (onClose) onClose()
+    },
+    [onFocus, onClose]
+  )
+
+  const resolvePathTitle = (path: Path) => {
+    const fields = documentType && documentType.fields
+    const field = fields && fields.find((curr) => curr.name === path[0])
+
+    return (field && field.type.title) || ''
+  }
+
+  const hasErrors = errors.length > 0
+  const hasWarnings = warnings.length > 0
+
+  if (!hasErrors && !hasWarnings) {
+    return null
+  }
+
+  return (
+    <ListContainer data-kind={kind}>
+      {hasErrors &&
+        errors.map((error, i) => (
+          <ListItem
+            // eslint-disable-next-line react/no-array-index-key
+            key={i}
```

Comment:
Is it possible to use something else as a `key` here, instead of an index?

Reply:
probably! but it's old code 😬